google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -1327,4 +1329,59 @@ public void floatIncrement() throws ExecutionException, InterruptedException {
       this.error = error;
     }
   }
+
+  @Test
+  public void getAllWithObserver() throws Exception {
+    DocumentReference ref = randomColl.document("doc1");
+    ref.set(ALL_SUPPORTED_TYPES_MAP).get();
+
+    final List<DocumentSnapshot> documentSnapshots = new ArrayList<>();
+    final DocumentReference[] documentReferences = {ref};
+    firestore.getAll(
+        FieldMask.of("foo"),
+        new ApiStreamObserver<DocumentSnapshot>() {
+
+          @Override
+          public void onNext(DocumentSnapshot documentSnapshot) {
+            documentSnapshots.add(documentSnapshot);
+          }
+
+          @Override
+          public void onError(Throwable throwable) {
+            Assert.fail("Error occurred while streaming");
+          }
+
+          @Override
+          public void onCompleted() {}
+        },
+        documentReferences);
+    Thread.sleep(1000);
+    assertEquals(map("foo", "bar"), documentSnapshots.get(0).getData());
+  }
+
+  @Test
+  public void getWithObserver() throws Exception {
+    final DocumentReference ref = randomColl.document("doc1");
+    final List<DocumentSnapshot> documentSnapshots = new ArrayList<>();
+    ref.set(ALL_SUPPORTED_TYPES_MAP).get();
+    ApiStreamObserver responseObserver =
+        new ApiStreamObserver<DocumentSnapshot>() {
+
+          @Override
+          public void onNext(DocumentSnapshot documentSnapshot) {
+            documentSnapshots.add(documentSnapshot);
+          }
+
+          @Override
+          public void onError(Throwable throwable) {
+            Assert.fail("Error occurred while streaming");
+          }
+
+          @Override
+          public void onCompleted() {}
+        };
+    ref.get(FieldMask.of("foo"), responseObserver, ref);
```

Comment:
I think this test will likely go away as per my previous suggestion of removing this API.